server/src/main/java/keywhiz/service/daos/SecretSeriesDAO.java

```diff
@@ -221,18 +240,23 @@ public int setCurrentVersion(long secretId, long secretContentId, String updater
 
     return dslContext.update(SECRETS)
         .set(SECRETS.CURRENT, secretContentId)
+        .set(SECRETS.EXPIRY, r.getExpiry())
         .set(SECRETS.UPDATEDBY, updater)
         .set(SECRETS.UPDATEDAT, now)
         .where(SECRETS.ID.eq(secretId))
         .execute();
   }
 
   public Optional<SecretSeries> getSecretSeriesById(long id) {
-    SecretsRecord r =
-        dslContext.fetchOne(SECRETS, SECRETS.ID.eq(id).and(SECRETS.CURRENT.isNotNull()));
+    SecretsRecord r = getSecretSeriesRecordById(id);
     return Optional.ofNullable(r).map(secretSeriesMapper::map);
   }
 
+  @VisibleForTesting
+  SecretsRecord getSecretSeriesRecordById(long id) {
+    return dslContext.fetchOne(SECRETS, SECRETS.ID.eq(id).and(SECRETS.CURRENT.isNotNull()));
+  }
```

Comment:
So far I've chosen not to expose the series-level expiration in the Java model (SecretSeries), since it's more of an internal optimization for the database. As a result, in order to examine the series-level expiration we need access to the database model's record object, which we previously didn't have.